include/sddf/blk/queue.h

```diff
@@ -54,26 +57,24 @@ typedef struct blk_request {
 /* Response struct contained in response queue */
 typedef struct blk_response {
     blk_response_status_t status; /* response status */
-    uintptr_t addr; /* encoded dma address of data */
-    uint16_t count; /* number of blocks allocated for corresponding request */
     uint16_t success_count; /* number of blocks successfully read/written */
     uint32_t id; /* stores corresponding request ID */
 } blk_response_t;
 
 /* Circular buffer containing requests */
 typedef struct blk_req_queue {
-    uint32_t write_idx;
-    uint32_t read_idx;
-    uint32_t size; /* number of slots in request queue */
+    uint32_t head;
+    uint32_t tail;
+    uint32_t size;
     bool plugged; /* prevent requests from being dequeued when plugged */
     blk_request_t buffers[BLK_REQ_QUEUE_SIZE];
 } blk_req_queue_t;
 
 /* Circular buffer containing responses */
 typedef struct blk_resp_queue {
-    uint32_t write_idx;
-    uint32_t read_idx;
-    uint32_t size; /* number of slots in response queue */
+    uint32_t head;
+    uint32_t tail;
+    uint32_t size;
```

Comment:
Can you make a note somewhere that the `size` field should not be stored in shared memory since it can be manipulated by the client which can lead to out of bounds accesses etc since it can also play with the head and tail indexes.